### ISSUE TITLE
Port entity locking feature from LWCEntityLocking fork.

### ISSUE THREAD

**Issue by mibby** [pop4959/LWCX#193]
**LWC Version (/lwc admin version):** 2.2.8-dev-70d605a (Dev 90 https://ci.codemc.io/view/All/job/pop4959/job/LWCX-dev/90/)
**Server version (/version):** Purpur dev 1612 (git-Purpur-1612) (Paper 1.18.2)

**Short five word summary of issue:** 
Entities such as armor stands and item frames cannot be locked.

**Detailed description of problem:**
I am migrating from the LWCEntityLocking fork by Brokkonaut.
https://github.com/Brokkonaut/LWCEntityLocking

This fork contains functionality for locking entities such as armor stands, item frames, paintings, chest minecarts, normal mobs, etc. It handles configuration similar to block locking with new sections under the protections config section.

i.e.
```
protections:
    entities:
        armor_stand:
            enabled: true
        painting:
            enabled: true
        item_frame:
            enabled: true
        chest_minecart:
            enabled: true
    special:
        noaimob:
            enabled: false
```

Adding these items to LWCX's protection config does not allow locking of the items.
```
    blocks:
        armor_stand:
            enabled: true
        painting:
            enabled: true
        item_frame:
            enabled: true
        chest_minecart:
            enabled: true
```

Would it be possible to pull in changes from that fork to allow locking of those entity types? It is licensed under BSD 2-Clause which allows modification and redistribution. My players relied heavily on being able to lock armor stands and item frames. The implementation in that fork worked relatively fine and bug free. The reason for migrating to LWCX was for the addon support & custom features such as ``/cdisplay`` for lecterns.

**Comment by pop4959** [comment]
This fork is a fork of LWCEntityLocking by Me_Goes_Rawr, so it already has these features. By the looks of it your config is incorrect as the entity "blocks" are not in all uppercase. This is how it has been designed and it looks like the fork you have been using modified it to allow lowercase as well.

I am currently not looking into adding any additional features into the existing entity locking implementation, as I have plans to completely overhaul this system in the future which should fix most if not all of the bugs and odd behaviors that exist currently.

If you have any other questions about this feel free to join the Discord as well. I usually respond there quicker than on GitHub.

**Comment by mibby** [comment]
I see. So uppercasing the material name of entities would allow locking of those "block" tiles / entities? Kind of weird that the entities would be case sensitive while normal chests and such aren't.

**Comment by pop4959** [comment]
AFAIK it is intentional to distinguish from actual blocks. Really logically they shouldn't even be in the same config section.

**Comment by mibby** [comment]
Confirmed uppercasing material names has it working, thanks! The only issue I could identify is action bar messages when breaking entities (armorstand) not showing name of item unregistered & breaking item frames showing the item frame is locked instead of unregistered when smacked. As long as they can be locked and unlocked though, those are the two important ones.

The fork I used did have the items identifiable in separate entities and special sections, so perhaps could be used as reference whenever the existing entity locking implementation is overhauled.
